ext/ITensorNetworksEinExprsExt/src/ITensorNetworksEinExprsExt.jl

```diff
@@ -0,0 +1,67 @@
+module ITensorNetworksEinExprsExt
+
+using ITensors
+using ITensorNetworks
+using ITensorNetworks: Index
+using EinExprs: EinExprs, EinExpr, einexpr, SizedEinExpr
+
+function EinExprs.einexpr(tn::ITensorNetwork; optimizer::EinExprs.Optimizer)
+  IndexType = Any
+  T = vertextype(tn)
+
+  tensors = EinExpr{IndexType}[]
+  tensor_map = Dict{Set{IndexType},T}()
+  sizedict = Dict{IndexType,Int}()
+
+  for (key, tensor) in pairs(vertex_data(tn))
+    _inds = collect(inds(tensor))
+    push!(tensors, EinExpr{IndexType}(; head=_inds))
+    tensor_map[Set(_inds)] = key
+    merge!(sizedict, Dict(_inds .=> size(tensor)))
+  end
+
+  _openinds = collect(externalinds(tn))
+  expr = SizedEinExpr(sum(tensors; skip=_openinds), sizedict)
+
+  return einexpr(optimizer, expr)
+end
+
+function ITensorNetworks.contraction_sequence(
+  alg::ITensorNetworks.Algorithm"einexpr", tn::Vector{ITensor}; kwargs...
+)
+  return contraction_sequence(alg, ITensorNetwork(tn); kwargs...)
+end
+
+function ITensorNetworks.contraction_sequence(
+  ::ITensorNetworks.Algorithm"einexpr",
+  tn::ITensorNetwork{T};
+  optimizer=EinExprs.Exhaustive(),
+)
+  IndexType = Any
+  T = vertextype(tn)
+
+  tensors = EinExpr{IndexType}[]
+  tensor_map = Dict{Set{IndexType},T}()
+  sizedict = Dict{IndexType,Int}()
+
+  for (key, tensor) in pairs(vertex_data(tn))
```

Comment:
Same as above in terms of naming variables and iterating over vertices and tensors.